examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,266 @@
+# Output every time step.
+#
+
+simulation.output_writers[:growth] =
+    NetCDFOutputWriter(
+        model,
+        (perturbation_norm = perturbation_norm,),
+        filepath=joinpath(@__DIR__, "perturbation_norm_shallow_water.nc"),
+        schedule=IterationInterval(1),
+        dimensions=(perturbation_norm=(),),
+        mode = "c")
+
+run!(simulation)
+
+# ## Visualize the results
+#
+
+using NCDatasets
+nothing # hide
+
+# Define the coordinates for plotting
+
+xf = xnodes(ω)
+yf = ynodes(ω)
+nothing # hide
+
+# Define keyword arguments for plotting the contours
+
+kwargs = (
+         xlabel = "x",
+         ylabel = "y",
+           fill = true,
+         levels = 20,
+      linewidth = 0,
+          color = :balance,
+       colorbar = true,
+           ylim = (-Ly/2, Ly/2),
+           xlim = (0, Lx)
+)
+nothing # hide
+
+# Read in the `output_writer` for the two-dimensional fields
+# and then create an animation showing both the total and perturbation
+# vorticities.
+
+using Plots
+
+ds = NCDataset(simulation.output_writers[:fields].filepath, "r")
+
+iterations = keys(ds["time"])
+
+anim = @animate for (iter, t) in enumerate(ds["time"])
+     ω = ds["ω"][:, :, 1, iter]
+    ωp = ds["ωp"][:, :, 1, iter]
+
+     ω_max = maximum(abs, ω)
+    ωp_max = maximum(abs, ωp)
+
+     plot_ω = contour(xf, yf, ω',  clim=(-1,  1),  title=@sprintf("Total ω at t = %.3f", t); kwargs...)
+    plot_ωp = contour(xf, yf, ωp', clim=(-ωp_max, ωp_max), title=@sprintf("Perturbation ω at t = %.3f", t); kwargs...)
+
+    plot(plot_ω, plot_ωp, layout = (1,2), size=(1200, 500))
+end
+
+close(ds)
+
+mp4(anim, "Bickley_Jet_ShallowWater.mp4", fps=15)
+
+# Read in the `output_writer` for the scalar field.
+
+ds2 = NCDataset(simulation.output_writers[:growth].filepath, "r")
+
+iterations = keys(ds2["time"])
+
+     t = ds2["time"][:]
+norm_v = ds2["perturbation_norm"][:]
+
+close(ds2)
+
+# We import the `fit` function from `Polynomials.jl` to compute the best-fit slope of the 
+# perturbation norm on a logarithmic plot. This slope corresponds to the growth rate.
+
+using Polynomials: fit
+
+I = 6000:7000
+
+degree = 1
+linear_fit_polynomial = fit(t[I], log.(norm_v[I]), degree)
+
+best_fit = exp.(linear_fit_polynomial[0] .+ linear_fit_polynomial[1] * t)
+
+plt = plot(t, norm_v,
+            yaxis = :log,
+            ylims = (1e-3, 30),
+               lw = 4,
+            label = "norm(v)", 
+           xlabel = "time",
+           ylabel = "norm(v)",
+            title = "growth of perturbation norm",
+           legend = :bottomright)
+plot!(plt, t[I], 2 * best_fit[I], # factor 2 offsets our fit from the curve for better visualization
+               lw = 4,
+            label = "best fit")
+            
+# We can compute the slope of the curve on a log scale, which approximates the growth rate
+# of the simulation. This should be close to the theoretical prediction.
+
+println("Growth rate in the simulation is approximated to be ", best_fit[1], ",\n",
```

Comment:
```suggestion
println("Growth rate in the simulation is approximated to be ", linear_fit_polynomial[1], ",\n",
```

Reply:
best_fit[1] is just the first point in the exponential fit.... `7.98673556281449e-5`...

Reply:
That makes more sense